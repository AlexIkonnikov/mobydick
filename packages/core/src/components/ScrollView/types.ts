import {ScrollViewProps as DefaultScrollViewProps} from 'react-native';

export type ScrollViewProps = DefaultScrollViewProps;
