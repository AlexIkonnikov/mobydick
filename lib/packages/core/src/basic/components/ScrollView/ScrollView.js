"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const react_native_1 = require("react-native");
const ScrollView = (0, react_1.forwardRef)((props, ref) => {
    return <react_native_1.ScrollView ref={ref} {...props}/>;
});
exports.default = ScrollView;
