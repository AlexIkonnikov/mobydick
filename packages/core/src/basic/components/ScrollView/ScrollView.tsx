import {forwardRef} from 'react';
import {ScrollView as DefaultScrollView} from 'react-native';

import {IScrollView, IScrollViewProps} from './types';

const ScrollView = forwardRef<IScrollView, IScrollViewProps>((props, ref) => {
  return <DefaultScrollView ref={ref} {...props} />;
});
export default ScrollView;
