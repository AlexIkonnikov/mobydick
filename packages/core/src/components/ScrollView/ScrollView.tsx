import React, {FC} from 'react';
import {ScrollView as DefaultScrollView} from 'react-native';

import {ScrollViewProps} from './types';

const ScrollView: FC<ScrollViewProps> = props => {
  return <DefaultScrollView {...props} />;
};
export default ScrollView;
