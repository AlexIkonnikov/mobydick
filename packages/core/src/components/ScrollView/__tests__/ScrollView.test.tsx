import React from 'react';
import {render} from '@testing-library/react-native';

import ScrollView from '../index';

describe('@npm/mobydick-core/ScrollView', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<ScrollView />);
    expect(toJSON()).toMatchSnapshot();
  });
});
