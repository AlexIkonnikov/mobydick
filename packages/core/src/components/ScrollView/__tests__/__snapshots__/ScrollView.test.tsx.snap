// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mobydick/core/ScrollView renders correctly 1`] = `
<RCTScrollView>
  <View />
</RCTScrollView>
`;
